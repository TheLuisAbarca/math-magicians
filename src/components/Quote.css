@import url("https://fonts.googleapis.com/css2?family=Lora:ital@1&display=swap");

.quote-section {
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  gap: 40px;
  padding: 10%;
  background-repeat: no-repeat;
  background-size: 160px;
}

.quote {
  font-size: 30px;
  color: #212121;
  font-family: "Lora", serif;
}

.author {
    font-size: 30px;
    color: #212121;
    font-family: "Lora", serif;
    font-weight: bold;
    font-style: italic;
  }

.line {
  height: 4px;
  background-color: #ff8a3d;
  width: 30%;
  align-self: center;
}