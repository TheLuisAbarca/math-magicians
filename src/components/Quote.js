import './Quote.css';

const famousQuote = 'Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.';
const authorQuote = '- William Paul Thurston';

const Quote = () => (
  <section className="quote-section">
    <div className="line" />
    <blockquote>
      <p className="quote">
        {famousQuote}
      </p>
      <cite className="author">
        {authorQuote}
      </cite>
    </blockquote>
  </section>
);

export default Quote;
