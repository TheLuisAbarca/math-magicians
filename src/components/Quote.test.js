import React from 'react';
import renderer from 'react-test-renderer';
import Quote from './Quote';

describe('App Component: Quote', () => {
    test('Verify Component is correctly rendered.', () => {
      const quoteComponent = renderer.create(<Quote />).toJSON();
      expect(quoteComponent).toMatchSnapshot();
    });
});