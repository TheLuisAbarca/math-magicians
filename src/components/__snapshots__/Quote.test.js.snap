// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component: Quote Verify Component is correctly rendered. 1`] = `
<section
  className="quote-section"
>
  <div
    className="line"
  />
  <blockquote>
    <p
      className="quote"
    >
      Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
    </p>
    <cite
      className="author"
    >
      - William Paul Thurston
    </cite>
  </blockquote>
</section>
`;
